thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineJob.java

```diff
@@ -56,9 +57,10 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
     Long id = getIdFromJobKey(jobKey.getName());
     DetectionConfigDTO configDTO = detectionDAO.findById(id);
 
-    // Make sure start time is not out of DETECTION_TASK_MAX_LOOKBACK_WINDOW
+    // Get the time window for the detection job.
+    long delay = getDetectionExpectedDelay(configDTO);
     long end = System.currentTimeMillis();
-    long start = Math.max(configDTO.getLastTimestamp(), end  - ThirdEyeUtils.DETECTION_TASK_MAX_LOOKBACK_WINDOW);
+    long start = Math.max(configDTO.getLastTimestamp(), end - ThirdEyeUtils.DETECTION_TASK_MAX_LOOKBACK_WINDOW - delay);
```

Comment:
Shall we change it in the `DataAvailabilityTaskScheduler` too?

Reply:
Good point. I am looking into that.

Reply:
I almost finished this change in `DataAvailabilityTaskScheduler` and am sending a PR on this.

Reply:
I am thinking of we can simplify this logic. Let's sync offline.